Skyra/Database/Models/Guild.cs

```diff
@@ -0,0 +1,361 @@
+	}
+
+	public class GuildMessage
+	{
+		public string? Farewell { get; set; } = null;
+		public string? Greeting { get; set; } = null;
+		public string? JoinDm { get; set; } = null;
+		public ulong[] IgnoredChannels { get; set; } = new ulong[0];
+		public bool ModerationDm { get; set; } = false;
+		public bool ModerationReasonDisplay { get; set; } = true;
+		public bool ModerationMessageDisplay { get; set; } = true;
+		public bool ModerationAutoDelete { get; set; } = false;
+		public bool ModeratorNameDisplay { get; set; } = true;
+
+		public ulong GuildId { get; set; }
+		public virtual Guild Guild { get; set; }
+	}
+
+	public struct GuildRolesAutomatic
+	{
+		[JsonProperty("role_id")]
+		public ulong RoleId { get; set; }
+
+		[JsonProperty("points")]
+		public ulong Points { get; set; }
+	}
+
+	public struct GuildRolesReaction
+	{
+		[JsonProperty("role_id")]
+		public ulong RoleId { get; set; }
+
+		[JsonProperty("channel_id")]
+		public ulong ChannelId { get; set; }
+
+		[JsonProperty("message_id")]
+		public ulong MessageId { get; set; }
+	}
+
+	public struct GuildRolesRoleSet
+	{
+		[JsonProperty("name")]
+		public string Name { get; set; }
+
+		[JsonProperty("roles")]
+		public ulong Roles { get; set; }
+	}
+
+	public class GuildRole
+	{
+		public string Administrator { get; set; }
+		public GuildRolesAutomatic[] Automatic { get; set; }
+		public ulong Initial { get; set; }
+		public GuildRolesReaction[] MessageReactions { get; set; }
+		public ulong Moderator { get; set; }
+		public ulong Muted { get; set; }
+		public ulong RestrictedReaction { get; set; }
+		public ulong RestrictedEmbed { get; set; }
+		public ulong RestrictedAttachment { get; set; }
+		public ulong RestrictedVoice { get; set; }
+		public ulong[] Public { get; set; }
+		public bool RemoveInitial { get; set; }
+		public ulong Dj { get; set; }
+		public ulong Subscriber { get; set; }
+		public GuildRolesRoleSet[] UniqueRoleSets { get; set; }
+
+		public ulong GuildId { get; set; }
+		public virtual Guild Guild { get; set; }
+	}
+
+	public class GuildAutoModeration
+	{
+		public ulong[] IgnoredRoles { get; set; } = new ulong[0];
+		public ulong[] IgnoredChannels { get; set; } = new ulong[0];
+
+		public GuildModerationAttachment Attachment { get; set; }
+		public GuildModerationCapital Capital { get; set; }
+		public GuildModerationLink Link { get; set; }
+		public GuildModerationMessage Message { get; set; }
+		public GuildModerationNewLine NewLine { get; set; }
+		public GuildModerationInvites Invites { get; set; }
+		public GuildModerationWord Word { get; set; }
+		public GuildModerationReaction Reaction { get; set; }
+		public GuildModerationNoMentionSpam NoMentionSpam { get; set; }
+
+		public ulong GuildId { get; set; }
+		public virtual Guild Guild { get; set; }
+	}
+
+	public enum GuildModerationSoftAction
+	{
+		None
+	}
+
+	public enum GuildModerationHardAction
+	{
+		None
+	}
+
+	public abstract class GuildModerationShared
```

Comment:
perhaps consider GuildModerationBase as a name - seems to make more sense to me.